openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -4,19 +4,35 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"time"
 
+	"golang.org/x/sync/singleflight"
 	"k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/cel/library"
+	"k8s.io/apiserver/pkg/warning"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/lru"
+
+	"github.com/google/cel-go/checker"
 
 	configv1 "github.com/openshift/api/config/v1"
+	authenticationcel "k8s.io/apiserver/pkg/authentication/cel"
 	crvalidation "k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
 )
 
 const PluginName = "config.openshift.io/ValidateAuthentication"
 
+const (
+	individualExpressionExcessiveCostThreshold = 10000000
+	fixedCost                                  = 1 << 20
```

Comment:
Let's add some comments about the reasoning -- this is the default apiserver limit on header size in bytes. We can make pessimistic size estimates by assuming that inputs were decoded from base64-encoded JSON and it's not going to be very precise, but it's still valuable to be clear about the distinction between "bytes of base64-encoded JSON", size (string length, list/map element count), and cost.